pkg/config/mesh/mesh.go

```diff
@@ -141,6 +141,11 @@ func DefaultMeshConfig() *meshconfig.MeshConfig {
 // will not be modified.
 func ApplyProxyConfig(yaml string, meshConfig *meshconfig.MeshConfig) (*meshconfig.MeshConfig, error) {
 	mc := proto.Clone(meshConfig).(*meshconfig.MeshConfig)
+	if mc == nil {
```

Comment:
how does clone return `nil`? is `meshConfig` already nil? If so, seems like the check should be before clone?

Also it seems odd that we will make a empty mesh config. in 99% of places we use DefaultMeshConfig.

This seems like we should be changing the place that _calls_ ApplyProxyConfig, instead of ApplyProxyConfig itself?